lmi_utils/gadget_utils/gadget_surface_utils.py

```diff
@@ -76,12 +76,13 @@ def pkl_2_pcd(self, source_path, destination_path):
             np_x = numpy.empty(shape[0]*shape[1])
             np_y = numpy.empty(shape[0]*shape[1])
             i = 0
-            for y in range(0, shape[1]):
-                for x in range(0, shape[0]):
-                    np_x[i] = offset[0] + x * resolution[0]
-                    np_y[i] = offset[1] + y * resolution[1]
-                    np_z[i] = offset[2] + profile[x][y] * resolution[2]
-                    i += 1
+            for y in range(0, shape[0]):
+                for x in range(0, shape[1]):
+                    if profile[y][x] != -32768:
```

Comment:
np.empty creates uninitialized data. The if statement skips assigning data to some elements in the array. This leaves some garbage data in the array.